File: tm4e/src/main/java/org/eclipse/tm4e/core/internal/theme/ThemeRaw.java
/**
 * Copyright (c) 2015-2017 Angelo ZERR.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 * <p>
 * SPDX-License-Identifier: EPL-2.0
 * <p>
 * Contributors:
 * Angelo Zerr <angelo.zerr@gmail.com> - initial API and implementation
 */
package org.eclipse.tm4e.core.internal.theme;

import org.eclipse.tm4e.core.theme.IRawTheme;
import org.eclipse.tm4e.core.theme.IRawThemeSetting;
import org.eclipse.tm4e.core.theme.IThemeSetting;

import java.util.Collection;
import java.util.HashMap;

/**
 * fix : Support vsc theme and thTheme
 */
public class ThemeRaw extends HashMap<String, Object> implements IRawTheme, IRawThemeSetting, IThemeSetting {

    private static final long serialVersionUID = -3622927264735492387L;

    @Override
    public String getName() {
        return (String) super.get("name");
    }

    @Override
    public Collection<IRawThemeSetting> getSettings() {
        Collection<IRawThemeSetting> settings = ((Collection<IRawThemeSetting>) super.get("settings"));
        if (settings != null) {
            return settings;
        }
        return (Collection<IRawThemeSetting>) super.get("tokenColors");
    }

    @Override
    public Object getScope() {
        return super.get("scope");
    }

    @Override
    public IThemeSetting getSetting() {
        return (IThemeSetting) super.get("settings");
    }

    @Override
    public IRawThemeSetting getEditorSetting() {
        return (IRawThemeSetting) super.get("colors");
    }

    @Override
    public Object getFontStyle() {
        return super.get("fontStyle");
    }

    @Override
    public String getBackground() {
        return (String) super.get("background");
    }

    @Override
    public String getForeground() {
        return (String) super.get("foreground");
    }

}

File: editor/src/main/kotlin/com/dingyi/editor/language/textmate/theme/ThemeRaw.kt
package com.dingyi.editor.language.textmate.theme

import org.eclipse.tm4e.core.internal.theme.ThemeRaw
import org.eclipse.tm4e.core.theme.IRawThemeSetting

/**
 * @author: dingyi
 * @date: 2021/10/12 14:29
 * @description:
 **/
class ThemeRaw() : ThemeRaw() {


    override fun getSettings(): MutableCollection<IRawThemeSetting> {
        val superSettings = super.getSettings()
        if (superSettings != null) {
            return superSettings
        }
        return this["tokenColors"] as MutableCollection<IRawThemeSetting>
    }

    val editorSettings: Map<String, String>
        get() = this["colors"] as Map<String, String>

}